core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java

```diff
@@ -58,11 +59,40 @@ public void createTable(
       String namespace, String table, TableMetadata metadata, Map<String, String> options)
       throws ExecutionException {
     String fullKeyspace = fullKeyspace(namespace);
-    createKeyspaceIfNotExists(fullKeyspace, options);
     createTableInternal(fullKeyspace, table, metadata, options);
     createSecondaryIndex(fullKeyspace, table, metadata.getSecondaryIndexNames());
   }
 
+  @Override
+  public void createNamespace(String namespace, Map<String, String> options)
+      throws ExecutionException {
+    CreateKeyspace query =
```

Comment:
I think if the storage supports `ifNotExists` natively, we should use it when the `ifNotExists` flag is true.